footer: Integrate newly designed footer.

amanagr opened the pull request:
This footer was designed and mostly written by @terpimost. Original issue: #24082

This adds a new design of the footer for both corporate and not corporate enabled pages.

before:
<img width="1750" alt="Screenshot 2023-02-10 at 3 41 10 PM" src="https://user-images.githubusercontent.com/25124304/218065618-f4acca7c-f149-4125-9a4f-ec664986aacc.png">

<img width="1750" alt="image" src="https://user-images.githubusercontent.com/25124304/218066126-c1bed9c9-557a-4072-a92f-5d3fd4016c18.png">

after:
corporate disabled:
<img width="1036" alt="Screenshot 2023-02-10 at 1 23 14 PM" src="https://user-images.githubusercontent.com/25124304/218064583-97b1fc8e-15d5-4e3e-b623-22f4d89f0a6b.png">
corporate enabled:
<img width="1750" alt="Screenshot 2023-02-10 at 3 37 34 PM" src="https://user-images.githubusercontent.com/25124304/218064624-c9e8b167-42dc-45ea-b20e-c7797391a25f.png">


@timabbott splitting it into 2 parts was difficult since the links are very different from what they were before.

alya commented:
Thanks!

It looks like the "New organization" and "Log in" links don't go anywhere?

alya commented:
Let's drop the "Modifying Zulip" link.

amanagr commented:
@timabbott  test is failing due to this check `self._test("/hello/", "Chat for distributed teams", landing_missing_strings=["Log in"])`, so should we remove the `Log in` link from the footer ?

alya commented:
Let's change "Sponsorship" to "Support Zulip", and link it to `/help/support-zulip-project`.

alya commented:
Maybe we should use "Customer stories" rather than "Case studies"? I think it's more consistent with the top nav and the page itself.

alya commented:
Should the [Case studies](https://zulip.com/use-cases/) and [Open communities directory](https://zulip.com/communities/) links be relative, like all the rest?

alya commented:
Note that there is no longer a "Home" link here. Do we feel OK about that?

alya commented:
I think the Attributions links for the "corporate disabled" footer should go to zulip.com rather than being relative? We also don't actually need it at the moment, I guess, because the only image it has is on /for/business in the /corporate directory. But I think we have an open PR to use an image with an attribution on /integrations? I'm not finding it at the moment, but if that's correct, we are going to need it after all.

timabbott commented:
> @timabbott test is failing due to this check self._test("/hello/", "Chat for distributed teams", landing_missing_strings=["Log in"]), so should we remove the Log in link from the footer or add it conditionally similar to what we have in mind for the navbar?

The test is just a reflection of what the design concept was. What would the "Log in" link in the footer do? I think the reason it's not expected is because we don't have a realm to log into -- if it takes you to the "enter your realm" page, maybe that's OK.

timabbott commented on templates/zerver/footer.html:
```diff
@@ -1,186 +1,111 @@
+            <p>
+                <a href="/why-zulip/">{{ _("Why Zulip") }}</a>
+                <a href="/features/">{{ _("Features") }}</a>
+                <a href="/plans/">{{ _("Plans &amp; pricing") }}</a>
```
I would expect us to want a normal `&` here and below?

amanagr replied:
Okay, changed.

timabbott commented on templates/zerver/footer.html:
```diff
@@ -1,186 +1,111 @@
+        </div>
+        <div class="footer__section">
+            <span class="footer__section-title">
+                {{ _("ABOUT US") }}
```
Do we want these tagged for translation in all-caps form, or to use CSS to capitalize the section titles?

amanagr replied:
If it makes a difference I can change it but I don't think it should.

amanagr commented:
> Should the [Case studies](https://zulip.com/use-cases/) and [Open communities directory](https://zulip.com/communities/) links be relative, like all the rest?

No, I recall we decided to directly link them to zulip.com since the data for `/communities` is only available on zulip.com and I don't recall the reason for case studies.

amanagr commented:
> Note that there is no longer a "Home" link here. Do we feel OK about that?

I think it is fine since navbar will have the one.

amanagr commented:
I addressed all the comments in the code.

terpimost commented:
Sorry, I'm very confused which pages should work and shouldn't work. Could you please provide links to all kinds of pages with different footer?

I went to hello page and here is what I see 
<img width="1472" alt="image" src="https://user-images.githubusercontent.com/1903309/218643392-f9217433-592b-443b-8cd1-89f881e5c103.png">

amanagr commented:
My bad, all the pages should have the new footer, I converted all the hex values to hsl last second and that introduced this bug.

terpimost commented:
I guess my comments in a different PR should be here too? https://github.com/zulip/zulip/pull/24301#issuecomment-1434069224 
I still see problems on the /new page:
<img width="1539" alt="image" src="https://user-images.githubusercontent.com/1903309/219549908-299ca675-520d-4030-9550-766c4069a190.png">

amanagr commented:
I cannot reproduce it, did you fetch the new changes?

terpimost commented:
I managed to make it work. Here are the things I still see (not critical):

1.
http://localhost:9991/team/ : 
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/1903309/219901482-9d04e456-055e-4530-9b25-81f838b3f5bd.png">

2. white background if I scroll more down 

3. In the light theme I see dark underlines (should not be there): 
<img width="1366" alt="image" src="https://user-images.githubusercontent.com/1903309/219901623-b5eb45f6-c4f8-4d43-9261-bf14fe65226c.png">

4. Footer © has a different font, but it is not critical to use another font for a footer for 1 line of text. So lets just make it a proper os3 font: 
<img width="403" alt="image" src="https://user-images.githubusercontent.com/1903309/219901816-34b819f6-0eae-4a2e-89e3-4886f3e69a86.png">

5. Please remove unnecessary border for the header since it is adding vertical height... you can remove bottom margin too:
<img width="1691" alt="image" src="https://user-images.githubusercontent.com/1903309/219901867-8aa4c7c0-e1f8-420a-b9f0-86e71c1f6df1.png">

6. I suggest to rename `Open communities directory` to `Open communities`, otherwise it adds horizontal width to the whole column
<img width="547" alt="image" src="https://user-images.githubusercontent.com/1903309/219901943-9b1f2634-6f73-4d7b-a5e0-6ecb60333810.png">

amanagr commented:
@terpimost addressed all your comments except the 4th one. What is an os3 font?

terpimost commented:
@amanagr sorry, I I meant source sans 3. In original design it was open Sans just for that ©

amanagr commented:
Okay, that seems to be the default.

alya commented:
Looks good to me on the landing pages!

I'm finding the new footer a bit distracting on pages like `/accounts/go/` and `/new`. It's possible that the right solution is to have a minimal footer on those pages, rather than the full set of links.

alya commented:
@timabbott A note for you to keep in mind: this drops the "Home" link to `zulip.com` from the footer.

terpimost commented:
> It's possible that the right solution is to have a minimal footer on those pages, rather than the full set of links.

I suggest we do that as a separate issue. Requires thoughts on what to include and design...

terpimost commented:
> finding the new footer a bit distracting on pages like `/accounts/go/` and `/new`

A temp solution could be make a layout in a way that footer is always a second screen, so content part is always 100vh.

alya commented:
> > finding the new footer a bit distracting on pages like `/accounts/go/` and `/new`
> 
> A temp solution could be make a layout in a way that footer is always a second screen, so content part is always 100vh.

Yeah, that might help!

timabbott commented on web/src/portico/team.js:
```diff
@@ -67,7 +67,9 @@ export default function render_tabs() {
-    const contributors_list = page_params.contributors.filter((c) => exclude_bot_contributors(c));
+    const contributors_list = page_params.contributors
+        ? []
+        : page_params.contributors.filter((c) => exclude_bot_contributors(c));
```
Hmm, is this check inverted though?

Is the list empty, or are we failing to pass it in development? Might be the latter is the issue.

amanagr replied:
Yeah, my bad. This is just not working in development.

timabbott commented on templates/zerver/footer.html:
```diff
@@ -1,186 +1,108 @@
+            </p>
+            <p>
+                <a href="https://zulip.com/use-cases/">{{ _("Customer stories") }}</a>
+                <a href="https://zulip.com/communities/">{{ _("Open communities") }}</a>
```
These links look to have been accidentally converted from relative to hardcoding zulip.com.

amanagr replied:
Yeah, for some reason I intentionally converted them but not sure why.

timabbott commented on templates/zerver/footer.html:
```diff
@@ -1,186 +1,108 @@
+            {% endif %}
+            <a href="{{ root_domain_uri }}/terms/">{{ _("Terms of Service") }}</a>
+            <a href="{{ root_domain_uri }}/privacy/">{{ _("Privacy policy") }}</a>
+            <a href="https://zulip.com/attribution/">{{ _("Attributions") }}</a>
```
We should rename this back to "Website attributions"; they're not for the product, just the website.

amanagr replied:
Okay.

timabbott commented on templates/zerver/footer.html:
```diff
@@ -1,186 +1,108 @@
-            </li>
-        </ul>
+    {% endif %}
+    <div class="footer__legal {% if not corporate_enabled %}footer__legal_not_corporate{% endif %}">
```
I felt like there was too much space at the bottom, with 28px bottom margin for footer section paragraphs, plus 24px top margin for this legal footer, resulting in a really tall empty space above the terms/privacy section:

![image](https://user-images.githubusercontent.com/2746074/222832804-e4c61a43-b6f4-4172-a9fc-b62fc43327a0.png)

Removing that 24px top margin results in this:

![image](https://user-images.githubusercontent.com/2746074/222832679-e40ec769-4237-4ceb-ab06-e142fe0b51c8.png)

I think that's probably what is intended (i.e. it was basically a bug that the 24px margin wasn't collapsed with the 28px margin), but @terpimost thoughts?

timabbott commented:
> A temp solution could be make a layout in a way that footer is always a second screen, so content part is always 100vh.

Yeah, I think this is a viable direction for the distraction issue. @amanagr can you try working on that in a separate PR? I think it has enough testing complexity (since it's like a dozen pages in the login, password reset, etc. flows) that it'd be better to have its own space. And I think while it'd be nice to merge together with this, I don't think it's a big deal to have the more visible footer for a week if that's how things end up in production.

I think the other sensible option would be to create a "only colors different" variable of the footer that uses a much less bold background color, like a light purple (but the exact same layout and general visuals), that we can use on these dozen transactional pages to still have the footer be available, but not have it jump out at you. The existing https://zulip.com/new doesn't have this distraction problem because the footer there is just really anti-attention-grabbing.

There's also the option of just not displaying the full footer at all on these pages. I'm not sure what's best, but I do want to know that it's pretty common for someone's first introduction to Zulip to be getting an email invitation to join and clicking the link and landing on the registration form with no intermediate pages that tell you what Zulip is, so I do think that what we have on the footer in at least that transactional page matters.

* One other issue is that I think we don't need "website attributions" on the self-hosted footer (corporate_disabled), and I'm worried that the dark purple is pretty intense contrast on non-landing pages  with just a thin line at the bottom: Here's an example:

![image](https://user-images.githubusercontent.com/2746074/222837880-173a183a-1f91-4e02-b58d-a339a3f5e416.png)

It doesn't look terrible to me but the self-hosted footers for these sites are definitely worth thinking about when addressing the above. (In particular, `100vh` is the wrong answer in that case since it'll create a scrollbar just hiding one line of footer)

amanagr commented on templates/zerver/footer.html:
```diff
@@ -1,186 +1,108 @@
+        </div>
+        <div class="footer__section">
+            <span class="footer__section-title">
+                {{ _("RESOURCES") }}
```
Ok, used same style for all.

amanagr commented on templates/zerver/footer.html:
```diff
@@ -1,186 +1,108 @@
+                <a href="/development-community/">{{ _("Development community") }}</a>
                 <a href="https://zulip.readthedocs.io/en/latest/translating/translating.html">
-                    {{ _("Translation") }}
+                {{ _("Translation") }}
```
@alya FYI

amanagr commented on templates/zerver/footer.html:
```diff
@@ -1,186 +1,108 @@
+            <span class="footer__section-title">
+                {{ _("ABOUT US") }}
+            </span>
+            <p>
```
Ok, done.

amanagr commented:
@timabbott updated! I will open a PR for your last comment.

alya commented:
Thanks! I think I'd like to see more white space above the footer on some landing pages (/for/X, /development-community are the ones I noted):

![Screen Shot 2023-03-08 at 11 09 18 AM](https://user-images.githubusercontent.com/2090066/223815589-7963e7c7-84e7-4394-aa08-dbf63473393a.png)
![Screen Shot 2023-03-08 at 11 08 40 AM](https://user-images.githubusercontent.com/2090066/223815597-01fa6077-e7fb-4a22-963c-3e096443d876.png)

alya commented:
Looks good to me otherwise.

timabbott commented:
Pushed "About Us" => "About us" since the capitalization linter caught that violation of our sentence case policy.

timabbott commented:
I worked with @alya a bit on the self-hosted footer content and added the following changes.

```
diff --git a/templates/zerver/footer.html b/templates/zerver/footer.html
index 09851106bd..c8abd5032b 100644
--- a/templates/zerver/footer.html
+++ b/templates/zerver/footer.html
@@ -99,14 +99,20 @@
     {% endif %}
     <div class="footer__legal {% if not corporate_enabled %}footer__legal_not_corporate{% endif %}">
         <div class="footer__legal-container">
+            {% if corporate_enabled %}
             <div class="copyright">© Kandra Labs, Inc. (“Zulip”)</div>
+            {% else %}
+            <div class="copyright">Powered by <a href="https://zulip.com">Zulip</a></div>
+            {% endif %}
             <div class="footer__legal-spacer"></div>
             {% if not corporate_enabled %}
             <a href="/help">{{ _("Help center") }}</a>
             {% endif %}
             <a href="{{ root_domain_uri }}/terms/">{{ _("Terms of Service") }}</a>
             <a href="{{ root_domain_uri }}/privacy/">{{ _("Privacy policy") }}</a>
+            {% if corporate_enabled %}
             <a href="https://zulip.com/attribution/">{{ _("Website attributions") }}</a>
+            {% endif %}
         </div>
     </div>
 </footer>
```

I think this is now good to integrate in terms of content, but it needs:
* A translation tag on the "Powered by" line.
* Some CSS tweaks for mobile sizes; probably the first line should be centered and the there should be symmetric bottom margin with the top margin (there's too little on the bottom in this size).

![image](https://user-images.githubusercontent.com/2746074/223822179-908ef90d-12f9-408b-89f3-ee988ef7d126.png)

timabbott commented:
* I also noticed an unfortunate inconsistent between the underline placement for `:focus` and `:hover` CSS on links -- if you middle-click a link, the link becomes focused, and has a tightly placed underline. If you then hover over that link, the underline jumps like 10px down in a way that's distracting. I'm not really sure what the intent is, but probably we want the underline part of these CSS rules to be identical. @terpimost FYI in case you can clarify that -- I think that's the last blocker for integrating the new footer!

amanagr commented:
<img width="331" alt="Screenshot 2023-03-09 at 7 59 54 AM" src="https://user-images.githubusercontent.com/25124304/223913380-c62d141d-fdf7-4fcb-863c-50dd3299eb75.png">

Should be fixed now. Addressed all the issues noted above.

alya commented:
Hm, on /self-hosting, I think we should get rid of the white stripe above the footer:

<img width="1496" alt="Screen Shot 2023-03-08 at 10 39 38 PM" src="https://user-images.githubusercontent.com/2090066/223941349-c4359174-ffc9-42b9-9d57-9ce0f386fc8e.png">

alya commented:
I didn't see any other issues when poking around. Haven't tested the self-hosted footer.

amanagr commented:
Okay, then I only added the extra margin to develop-community and communities page.
<img width="776" alt="Screenshot 2023-03-09 at 7 01 20 PM" src="https://user-images.githubusercontent.com/25124304/224064645-251f5c75-271d-4e69-b8a2-a6130e30686a.png">
<img width="795" alt="Screenshot 2023-03-09 at 7 01 37 PM" src="https://user-images.githubusercontent.com/25124304/224064663-bdbdeb92-bd47-43b4-861d-b5ffc113d7ff.png">

timabbott commented:
This looks good enough to merge, thanks for all the work on this @amanagr! @terpimost FYI, we can now move onto integrating the next big chunk of the redesign :)

That said, I'm a bit puzzled why we need the `extra_margin_before_footer` rather than just having it be margin attached to `bottom-register-buttons` -- does /for/events look bad with such a margin?
```
$ git grep bottom-register-buttons
templates/corporate/development-community.html:        <div class="bottom-register-buttons extra_margin_before_footer">
templates/corporate/for/communities.html:    <div class="bottom-register-buttons extra_margin_before_footer">
templates/corporate/for/events.html:        <div class="bottom-register-buttons">
templates/corporate/for/open-source.html:    <div class="bottom-register-buttons extra_margin_before_footer">
templates/corporate/for/research.html:        <div class="bottom-register-buttons extra_margin_before_footer">
web/styles/portico/landing_page.css:    .bottom-register-buttons.extra_margin_before_footer {
web/styles/portico/landing_page.css:.bottom-register-buttons {
```

(But that seems easy to fix forward if it is indeed an improvement to do so).

Is there a #design thread that we should update?

alya commented:
> Is there a #design thread that we should update?

Nope. I don't think we need one?

amanagr commented:
I know this is not the smoothest thing to do here, but `extra_margin_before_footer` helps us fix the pages that I thought really required the extra margin. Also, it is limiting for `bottom-register-buttons` to have the extra margin attached to them so that they can be used more freely in other pages without disturbing the impact on old ones.
